### ISSUE TITLE
Autosubmit recovering incomplete logs

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#770]
In GitLab by @mcastril on Jan 14, 2022, 14:44

I have seen that on particular occasions, Autosubmit copies the job logfiles before they are written to the end. 

You can see some examples here:

`/esarchive/autosubmit/a48j/tmp/LOG_a48j`

`/esarchive/scratch/Earth/mcastril/a48j/LOG_a48j/`

The fix must not be an easy one, maybe we can copy the log and after some time verify if it changed.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 14, 2022, 14:53

This happened in an experiment where the cloning of the model is failing. The clone issue was first reported by @mgali and I could reproduce it.

```
Cloning into '.'...
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
```

Setting `git config --global http.postBuffer 524288000` works but the clone is extremely slow... Latency is somehow high at this moment too

FYI @kserradell

**Comment by kinow** [comment]
In GitLab by @kserradell on Jan 14, 2022, 14:55

Where are you running the clone?

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 14, 2022, 17:22

This one is in `bscesautosubmit01`. Auto-NEMO has an option to defer the clone to the cluster but in this case, it does it in the first job so the code is locally available for development.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 14, 2022, 17:26

Thanks for asking because I verified it and I was wrong. I thought that it was being checked out in `bscesautosubmit01` but actually we do it in the `000`. It has a point because it's a download and we didn't want to slowdown the AS VMs but now it makes sense that it goes so slow...

We can temporally move this workload to LOCAL (the AS VM where it's being executed).

**Comment by kinow** [comment]
In GitLab by @mcastril on Jan 14, 2022, 17:36

Much better now (it cloned both NEMO and XIOS in ~10 minutes).

@mgali you can change the platform where the LOCAL_SETUP is run to `LOCAL`

```
[LOCAL_SETUP]
FILE = templates/localsetup.tmpl.sh
PLATFORM = LOCAL
CHECK = true
```

Note: This is only appliable to auto-nemo because it performs the model clone `inside` the workflow

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 9, 2022, 18:51

mentioned in issue auto-ecearth3#1759

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 1, 2022, 15:02

@dbeltrankyl do you think that this is fixed after the changes to recover longer logfiles?

@ yruprich had this problem using 3.13.0. On the other hand, @pabretonniere told me that he had the same issue but it disappear when moving to 3.14.0b (after following @eferre1 advise).

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 1, 2022, 15:09

> @dbeltrankyl do you think that this is fixed after the changes to recover longer logfiles?

There was more than one issue related to logs fixed on 3.14.0b. I don't know if this issue was caused by the bug of log retrials jobs  or the long logs files mem issue but either way, I think it should've been fixed on 3.14.0b

**Comment by kinow** [comment]
In GitLab by @ yruprich on Sep 1, 2022, 18:44

mentioned in issue auto-ecearth3#1788
